src/Components/Web.Extensions/src/InputFile/IFileListEntry.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Represents the data of a file selected from an <see cref="InputFile"/> component.
+    /// </summary>
+    public interface IFileListEntry
```

Comment:
Can we call this `IBrowserFile`? The browser API is called `File`: https://developer.mozilla.org/en-US/docs/Web/API/File

Reply:
Is there a reason we have an interface and implementation? Do we expect users to implement this contract?

Reply:
I believe properties that can be deserialized from JSON must have public setters - correct me if I'm wrong on this. If that's the case, we probably don't need users to be able to tinker with the various properties, hence the internal `FileListEntry` and the public `IFileListEntry` that describes those properties as read-only.

Reply:
`IBrowserFile` sounds fine to me. In most cases people won't see the type name in their code anyway since they will receive the `InputFileChangeEventArgs` then just iterate over its `Files` collection.

> I believe properties that can be deserialized from JSON must have public setters - correct me if I'm wrong on this. If that's the case, we probably don't need users to be able to tinker with the various properties, hence the internal FileListEntry and the public IFileListEntry that describes those properties as read-only.

Yes, I agree. However if we wanted we could swap the "public interface, internal class" pair with a "public abstract class, internal concrete subclass" pair. That has the benefit of letting us add more members in the future non-breakingly.

Not a big deal though. Do either of you feel strongly either way?

Reply:
I'm not partial to keeping the interface - I'll go ahead and swap it for an abstract class.

Reply:
Actually, @SteveSandersonMS, I just want to clarify my understanding of what you had in mind. Is the change to enable us to add more properties to `IBrowserFile` without forcing our customers to modify their own types implementing `IBrowserFile`? If so, wouldn't default interface implementations solve this? The challenge with turning `IBrowserFile` into an abstract class is that we can't define properties in the abstract class as read-only but define them in the internal class as read-write. There are workarounds of course, but that seemed messier to me than using default interface implementation.